plugin.py

```diff
@@ -129,7 +181,165 @@ def run(self, edit: sublime.Edit) -> None:
+            if not shutil.which("cargo"):
+                sublime.error_message('Cannot find "cargo" on path.')
+                return
+            cargo_path = '"{}"'.format(shutil.which("cargo"))
+        command_to_run = [cargo_path] + output["args"]["cargoArgs"] + \
+            output["args"]["cargoExtraArgs"] + output["args"]["executableArgs"]
+        print(command_to_run)
+        cmd = " ".join(command_to_run)
+        args = {
+            "title": output["label"],
+            "shell_cmd": cmd,
+            "cwd": output["args"]["workspaceRoot"],
+            "auto_close": get_setting(self.view, "terminus_auto_close", False)
+        }
+        if get_setting(self.view, "terminus_use_panel", False):
+            args["panel_name"] = output["label"]
+        window.run_command("terminus_open", args)
+
+    def on_result(self, payload: Any) -> None:
+        raise NotImplementedError()
+
+
+class RustAnalyzerRunProject(RustAnalyzerExec):
+    session_name = "rust-analyzer"
+
+    def run(self, edit: sublime.Edit) -> None:
+        params = text_document_position_params(self.view, self.view.sel()[0].b)
+        session = self.session_by_name(self.session_name)
+        if session is None:
+            return
+        session.send_request(Request("experimental/runnables", params), self.on_result)
+
+    def on_result(self, payload: List[Runnable]) -> None:
+        items = [item["label"] for item in payload]
+        self.items = items
+        self.payload = payload
+        view = self.view
+        window = view.window()
+        if window is None:
+            return
+        sublime.set_timeout(
+            lambda: window.show_quick_panel(items, self.callback)
+        )
+
+    def callback(self, option: int) -> None:
+        if option == -1:
+            return
+        self.run_termius(self.items[option], self.payload)
+
+
+class RustAnalyzerCheckProject(RustAnalyzerExec):
+    session_name = "rust-analyzer"
+    check_phrase = "cargo check"
+
+    def run(self, edit: sublime.Edit) -> None:
+        session = self.session_by_name(self.session_name)
+        if session is None:
+            return
+        params = text_document_position_params(self.view, self.view.sel()[0].b)
+        session.send_request(Request("experimental/runnables", params), self.on_result)
+
+    def on_result(self, payload: List[Any]) -> None:
+        self.run_termius(self.check_phrase, payload)
+
+
+class RustAnalyzerTestProject(RustAnalyzerExec):
+    session_name = "rust-analyzer"
+    check_phrase = "cargo test"
+
+    def run(self, edit: sublime.Edit) -> None:
+        params = text_document_position_params(self.view, self.view.sel()[0].b)
+        session = self.session_by_name(self.session_name)
+        if session is None:
+            return
+        session.send_request(Request("experimental/runnables", params), self.on_result)
+
+    def on_result(self, payload: List[Runnable]) -> None:
+        self.run_termius(self.check_phrase, payload)
```

Comment:
Those are no longer ussed?

Reply:
Kept em just in case. I can remove

Reply:
- There is a lot of duplication here
- If the commands are not even documented, there is really no point in keeping them. Nobody will know that they exist.
- If there is really a desire to run those directly without them quick panel that that can be implemented using an optional argument to `RustAnalyzerRunProject`